cpp/src/utility/z_order.cuh

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+// adapted from https://dl.acm.org/doi/10.1109/TC.2007.70814
+__constant__ const uint16_t dilate_tab2[256] = {
```

Comment:
spell out "table". 
```suggestion
__constant__ const uint16_t dilate_table[256] = {
```

Also, consider `static __device__` for this rather than `__constant__`. The latter is optimized for uniform access within warps and thread blocks, and the serialization that results when that is not the case could be worse than the hierarchical caching of regular `__device__` memory.